pkg/build/controller/build/defaults/defaults.go

```diff
@@ -100,6 +100,12 @@ func (b BuildDefaults) applyPodDefaults(pod *v1.Pod) {
 			}
 		}
 	}
+
+	if pod.Spec.Tolerations == nil && len(b.config.Tolerations) != 0 {
```

Comment:
if pod.Spec.Tolerations is nil, you should be creating a new array and populating it.

Reply:
(actually I think w/ arrays you can just always append, you don't need to explicitly instantiate the empty array first)

Reply:
but regardless you should be adding the default tolerations to the array.  the defaulting should be done on a key by key basis.  Same way the annotations are being done currently.

Reply:
Nil and zero length are equivalent here, and you should check zero length